src/udev/udev-node.c

```diff
@@ -232,7 +228,39 @@ static int link_update(sd_device *dev, const char *slink, bool add) {
                                 r = -errno;
                 } while (r == -ENOENT);
 
-        return r;
+        for (i = 0; i < LINK_UPDATE_MAX_RETRIES; i++) {
+                struct stat st1 = {}, st2 = {};
+
+                r = stat(dirname, &st1);
+                if (r < 0 && errno != ENOENT)
+                        return -errno;
+
+                r = link_find_prioritized(dev, add, dirname, &target);
+                if (r < 0) {
+                        log_device_debug(dev, "No reference left, removing '%s'", slink);
+                        if (unlink(slink) == 0)
+                                (void) rmdir_parents(slink, "/");
+
+                        break;
+                } else {
+                        r = node_symlink(dev, target, slink);
+                        if (r < 0) {
+                                (void) unlink(filename);
+                                break;
+                        } else if (r == 1)
```

Comment:
maybe add a brief comment what r > 0 means, i.e `/* actually created a symlink, i.e. was not already in place */`